src/coreclr/debug/daccess/dacdbiimpl.cpp

```diff
@@ -7208,6 +7208,10 @@ HRESULT DacDbiInterfaceImpl::GetTypeLayout(COR_TYPEID id, COR_TYPE_LAYOUT *pLayo
     DD_ENTER_MAY_THROW;
 
     PTR_MethodTable mt = PTR_MethodTable(TO_TADDR(id.token1));
+
+    if (!mt->IsClassPointerValid())
+        return CORDBG_E_TARGET_INCONSISTENT;
```

Comment:
DacValidateMethodTable() might be better to validate the MT. It does a more through check and should be safe from AVs/SIGSEGV even though it calls things like GetClassWithPossibleAV and GetMethodTableWithPossibleAV, etc.